open_instruct/dataset_transformation.py

```diff
@@ -807,16 +807,19 @@ def load_or_transform_dataset(self, dcs: List[DatasetConfig], tc: TokenizerConfi
         print(f"Cache not found, transforming datasets...")
 
         # Transform each dataset
-        transformed_datasets = []
+        transformed_datasets = {}
         for dc in dcs:
             dataset = get_dataset_v1(dc, tc)
-            transformed_datasets.append(dataset)
+            split = dc.dataset_split
+            if split in transformed_datasets:
+                transformed_datasets[split] = concatenate_datasets([transformed_datasets[split], dataset])
+            else:
+                transformed_datasets[split] = dataset
```

Comment:
Here is a use case, say I want to mix "train, test, train, test" splits of 4 datasets. Would this implementation only load `"train, train"` of the first and thrid dataset? 



What do you think about making a default split like

https://github.com/allenai/open-instruct/blob/7d1c5d95cf59941283dd53f1c54c7f52452f6784/open_instruct/dataset_transformation.py#L801-L808

and 

https://github.com/allenai/open-instruct/blob/7d1c5d95cf59941283dd53f1c54c7f52452f6784/open_instruct/dataset_transformation.py#L858-L859

Reply:
Ah, I guess i was thinking of this differently, this setup makes more sense.

Reply:
Okay changed to this, and tested for my use case and it works!

Reply:
Did you push the commit?

Reply:
lol im dumb